test/cql-pytest/test_secondary_index.py

```diff
@@ -1749,3 +1749,46 @@ def test_index_filtering2_scan_and_per_partition_limit(cql, test_keyspace, use_i
         cql.execute(stmt, [1, 1, 0, 1])
         cql.execute(stmt, [1, 2, 0, 1])
         assert {(0, 1, 0, 1), (1, 1, 0, 1)} == set(cql.execute(f'SELECT * FROM {table} WHERE v=0 AND z=1 PER PARTITION LIMIT 1 ALLOW FILTERING'))
+
+# Test that when adding an index to a table, queries should not begin to
+# use it before it's fully built because that would yield wrong query
+# results. Reproduces issue #7963.
+# The test is marked cassandra_bug, because Cassandra 4 also fails on it -
+# the ALLOW FILTERING request right after the CREATE INDEX causes an
+# exception in SecondaryIndexManagement and a failed read. This is despite
+# CASSANDRA-8505 claiming that this issue was already fixed in 2015.
+@pytest.mark.xfail(reason="issue #7963")
+def test_unbuilt_index_not_used(cql, test_keyspace, cassandra_bug):
+    # The bigger "count" is the slower the test and the higher the chance
+    # of reproducing the bug #7963. With dev build on my laptop, count=100
+    # is enough for reproducing the failure in 90% of the runs.
+    count = 100
```

Comment:
You could use artificial delay via the error injector in the view builder to guarantee the desired effect.

Reply:
Ok, I don't insist, it was just a suggestion.